sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala

```diff
@@ -803,6 +803,21 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
     assertResult(Array(Row(1))) {
       sql("select 1 as a intersect select 1 as a").collect()
     }
+    assertResult(Array(Row(1))) {
+      sql("(select 1 as a) intersect (select 1 as a)").collect()
```

Comment:
Is it possible to test fix by adding these to the parser unit test without running the queries? 

As we develop more and more Spark features, I'd like to encourage doing actual unit tests or integration tests that run the minimal required code.